gridstatus/isone.py

```diff
@@ -406,40 +408,37 @@ def get_lmp(
 
         now = pd.Timestamp.now(tz=self.default_timezone)
         if market == Markets.REAL_TIME_5_MIN:
-            # todo handle intervals for current day
-            intervals = ["00-04", "04-08", "08-12", "12-16", "16-20", "20-24"]
+            intervals = self.lmp_real_time_intervals[:]
 
-            # optimze for current day
             if now.date() == date.date():
-                hour = now.hour
-                # select completed 4 hour intervals based on current hour
-                intervals = intervals[: math.ceil((hour + 1) / 4) - 1]
+                intervals = self._filter_intervals_in_range(
+                    date,
+                    end,
+                    self.lmp_real_time_intervals,
+                )
 
             dfs = []
             for interval in intervals:
                 msg = "Loading interval {}".format(interval)
                 log(msg, verbose=verbose)
                 u = f"https://www.iso-ne.com/static-transform/csv/histRpts/5min-rt-prelim/lmp_5min_{date_str}_{interval}.csv"  # noqa
-                dfs.append(
-                    pd.read_csv(
-                        u,
-                        skiprows=[0, 1, 2, 3, 5],
-                        skipfooter=1,
-                        engine="python",
-                    ),
-                )
+                try:
+                    dfs.append(
```

Comment:
This needs to be a `try...except` because the interval file may not be published for the provided `date` and `end`.